internal/pkg/service/stream/keboolasink/bridge/bridge_test.go

```diff
@@ -0,0 +1,214 @@
+package bridge_test
+
+import (
+	"bytes"
+	"context"
+	"testing"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/keboola/go-client/pkg/keboola"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.etcd.io/etcd/client/v3/concurrency"
+
+	deps "github.com/keboola/keboola-as-code/internal/pkg/service/common/dependencies"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/rollback"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/utctime"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/definition/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/dependencies"
+	bridgeTest "github.com/keboola/keboola-as-code/internal/pkg/service/stream/keboolasink/bridge/test"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/test"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdhelper"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdlogger"
+)
+
+func TestBridge_FullWorkflow(t *testing.T) {
+	t.Parallel()
+
+	// bgCtx - for internal background operations
+	bgCtx := context.Background()
+
+	clk := clock.NewMock()
+	clk.Set(utctime.MustParse("2000-01-01T01:00:00.000Z").Time())
+	by := test.ByUser()
+
+	// Fixtures
+	projectID := keboola.ProjectID(123)
+	branchKey := key.BranchKey{ProjectID: projectID, BranchID: 456}
+	sourceKey := key.SourceKey{BranchKey: branchKey, SourceID: "my-source"}
+	sinkKey := key.SinkKey{SourceKey: sourceKey, SinkID: "my-sink"}
+	ignoredKeys := etcdhelper.WithIgnoredKeyPattern("^definition/|storage/file/all/|storage/slice/all/|storage/volume/")
+
+	// Get services
+	d, mocked := dependencies.NewMockedLocalStorageScope(t, deps.WithClock(clk))
+	client := mocked.TestEtcdClient()
+	defRepo := d.DefinitionRepository()
+	storageRepo := d.StorageRepository()
+	fileRepo := storageRepo.File()
+	volumeRepo := storageRepo.Volume()
+
+	// Log etcd operations
+	var etcdLogs bytes.Buffer
+	rawClient := d.EtcdClient()
+	rawClient.KV = etcdlogger.KVLogWrapper(rawClient.KV, &etcdLogs, etcdlogger.WithMinimal())
+
+	// apiCtx - for operations triggered by an authorized API call
+	apiCtx := rollback.ContextWith(bgCtx, rollback.New(d.Logger()))
+	apiCtx = context.WithValue(apiCtx, dependencies.KeboolaProjectAPICtxKey, mocked.KeboolaProjectAPI())
+
+	// Register mocked responses
+	// -----------------------------------------------------------------------------------------------------------------
+	transport := mocked.MockedHTTPTransport()
+	{
+		bridgeTest.MockTokenStorageAPICalls(t, transport)
+		bridgeTest.MockBucketStorageAPICalls(t, transport)
+		bridgeTest.MockTableStorageAPICalls(t, transport)
+		bridgeTest.MockFileStorageAPICalls(t, clk, transport)
+	}
```

Comment:
So I suppose that this works the same as handler chain. We start by interacting with token API, receive token, use it to create Bucket, Table and finally File.

Reply:
:telephone_receiver: